Note that tests are failing due to this change:
```
ImportError while loading conftest '/__w/core/core/tests/components/hue/conftest.py'.
tests/components/hue/conftest.py:8: in <module>
    import aiohue.v1 as aiohue_v1
venv/lib/python3.9/site-packages/aiohue/__init__.py:4: in <module>
    from .v2 import HueBridgeV2  # noqa
venv/lib/python3.9/site-packages/aiohue/v2/__init__.py:15: in <module>
    from .controllers.config import ConfigController
venv/lib/python3.9/site-packages/aiohue/v2/controllers/config.py:6: in <module>
    from ...util import mac_from_bridge_id
venv/lib/python3.9/site-packages/aiohue/util.py:6: in <module>
    from types import UnionType
E   ImportError: cannot import name 'UnionType' from 'types' (/usr/local/lib/python3.9/types.py)
```

--- a/homeassistant/components/hue/manifest.json
+++ b/homeassistant/components/hue/manifest.json
@@ -3,7 +3,7 @@
   "name": "Philips Hue",
   "config_flow": true,
   "documentation": "https://www.home-assistant.io/integrations/hue",
-  "requirements": ["aiohue==4.1.1"],
+  "requirements": ["aiohue==4.1.2"],
   "ssdp": [
     {
       "manufacturer": "Royal Philips Electronics",

--- a/homeassistant/components/hue/manifest.json
+++ b/homeassistant/components/hue/manifest.json
@@ -3,7 +3,7 @@
   "name": "Philips Hue",
   "config_flow": true,
   "documentation": "https://www.home-assistant.io/integrations/hue",
-  "requirements": ["aiohue==4.1.0"],
+  "requirements": ["aiohue==4.1.1"],
   "ssdp": [
     {
       "manufacturer": "Royal Philips Electronics",

--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -191,7 +191,7 @@ aiohomekit==0.7.14
 aiohttp_cors==0.7.0
 
 # homeassistant.components.hue
-aiohue==4.1.1
+aiohue==4.1.2
 
 # homeassistant.components.homewizard
 aiohwenergy==0.8.0

--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -191,7 +191,7 @@ aiohomekit==0.7.14
 aiohttp_cors==0.7.0
 
 # homeassistant.components.hue
-aiohue==4.1.0
+aiohue==4.1.1
 
 # homeassistant.components.homewizard
 aiohwenergy==0.8.0

--- a/requirements_test_all.txt
+++ b/requirements_test_all.txt
@@ -141,7 +141,7 @@ aiohomekit==0.7.14
 aiohttp_cors==0.7.0
 
 # homeassistant.components.hue
-aiohue==4.1.0
+aiohue==4.1.1
 
 # homeassistant.components.homewizard
 aiohwenergy==0.8.0

--- a/requirements_test_all.txt
+++ b/requirements_test_all.txt
@@ -141,7 +141,7 @@ aiohomekit==0.7.14
 aiohttp_cors==0.7.0
 
 # homeassistant.components.hue
-aiohue==4.1.1
+aiohue==4.1.2
 
 # homeassistant.components.homewizard
 aiohwenergy==0.8.0

--- a/tests/components/hue/conftest.py
+++ b/tests/components/hue/conftest.py
@@ -8,7 +8,6 @@ from unittest.mock import AsyncMock, Mock, patch
 import aiohue.v1 as aiohue_v1
 import aiohue.v2 as aiohue_v2
 from aiohue.v2.controllers.events import EventType
-from aiohue.v2.models.clip import parse_clip_resource
 import pytest
 
 from homeassistant.components import hue
@@ -187,7 +186,7 @@ def create_mock_api_v2(hass):
 
     def emit_event(event_type, data):
         """Emit an event from a (hue resource) dict."""
-        api.events.emit(EventType(event_type), parse_clip_resource(data))
+        api.events.emit(EventType(event_type), data)
 
     api.load_test_data = load_test_data
     api.emit_event = emit_event

--- a/tests/components/hue/test_light_v2.py
+++ b/tests/components/hue/test_light_v2.py
@@ -97,8 +97,12 @@ async def test_light_turn_on_service(hass, mock_bridge_v2, v2_resources_test_dat
     assert mock_bridge_v2.mock_requests[0]["json"]["color_temperature"]["mirek"] == 300
 
     # Now generate update event by emitting the json we've sent as incoming event
-    mock_bridge_v2.mock_requests[0]["json"]["color_temperature"].pop("mirek_valid")
-    mock_bridge_v2.api.emit_event("update", mock_bridge_v2.mock_requests[0]["json"])
+    event = {
+        "id": "3a6710fa-4474-4eba-b533-5e6e72968feb",
+        "type": "light",
+        **mock_bridge_v2.mock_requests[0]["json"],
+    }
+    mock_bridge_v2.api.emit_event("update", event)
     await hass.async_block_till_done()
 
     # the light should now be on
@@ -186,7 +190,12 @@ async def test_light_turn_off_service(hass, mock_bridge_v2, v2_resources_test_da
     assert mock_bridge_v2.mock_requests[0]["json"]["on"]["on"] is False
 
     # Now generate update event by emitting the json we've sent as incoming event
-    mock_bridge_v2.api.emit_event("update", mock_bridge_v2.mock_requests[0]["json"])
+    event = {
+        "id": "02cba059-9c2c-4d45-97e4-4f79b1bfbaa1",
+        "type": "light",
+        **mock_bridge_v2.mock_requests[0]["json"],
+    }
+    mock_bridge_v2.api.emit_event("update", event)
     await hass.async_block_till_done()
 
     # the light should now be off
@@ -377,10 +386,20 @@ async def test_grouped_lights(hass, mock_bridge_v2, v2_resources_test_data):
         )
 
     # Now generate update events by emitting the json we've sent as incoming events
-    for index in range(0, 3):
-        mock_bridge_v2.api.emit_event(
-            "update", mock_bridge_v2.mock_requests[index]["json"]
-        )
+    for index, light_id in enumerate(
+        [
+            "02cba059-9c2c-4d45-97e4-4f79b1bfbaa1",
+            "b3fe71ef-d0ef-48de-9355-d9e604377df0",
+            "8015b17f-8336-415b-966a-b364bd082397",
+        ]
+    ):
+        event = {
+            "id": light_id,
+            "type": "light",
+            **mock_bridge_v2.mock_requests[index]["json"],
+        }
+        mock_bridge_v2.api.emit_event("update", event)
+    await hass.async_block_till_done()
     await hass.async_block_till_done()
 
     # the light should now be on and have the properties we've set
@@ -406,6 +425,12 @@ async def test_grouped_lights(hass, mock_bridge_v2, v2_resources_test_data):
     assert mock_bridge_v2.mock_requests[0]["json"]["on"]["on"] is False
 
     # Now generate update event by emitting the json we've sent as incoming event
+    event = {
+        "id": "f2416154-9607-43ab-a684-4453108a200e",
+        "type": "grouped_light",
+        **mock_bridge_v2.mock_requests[0]["json"],
+    }
+    mock_bridge_v2.api.emit_event("update", event)
     mock_bridge_v2.api.emit_event("update", mock_bridge_v2.mock_requests[0]["json"])
     await hass.async_block_till_done()
 

--- a/tests/components/hue/test_switch.py
+++ b/tests/components/hue/test_switch.py
@@ -69,7 +69,12 @@ async def test_switch_turn_off_service(hass, mock_bridge_v2, v2_resources_test_d
     assert mock_bridge_v2.mock_requests[0]["json"]["enabled"] is False
 
     # Now generate update event by emitting the json we've sent as incoming event
-    mock_bridge_v2.api.emit_event("update", mock_bridge_v2.mock_requests[0]["json"])
+    event = {
+        "id": "b6896534-016d-4052-8cb4-ef04454df62c",
+        "type": "motion",
+        **mock_bridge_v2.mock_requests[0]["json"],
+    }
+    mock_bridge_v2.api.emit_event("update", event)
     await hass.async_block_till_done()
 
     # the switch should now be off

